### PR TITLE
Start on a CSS pipeline with LESS

### PR DESCRIPTION
**Pull request by cheshire137**
This branch adds:
- Travis CI setup for tests and style checker
- LESS for styling
- Fixes style checker complaints
- Organizes code into src/ directory

Hmm, so the app starts, but the styling doesn’t seem to be applied. Any thoughts on this, @probablycorey?

### CODE REVIEW

File: .compilerc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (probablycorey) [inline .compilerc:1]: Is this a `electron-compile` thing? It's pretty nice.

File: src/index.html [hunk 2]
```diff
@@ -2,7 +2,8 @@
 <html>
   <head>
     <meta charset="UTF-8">
-    <title>Hello World!</title>
+    <title>GitHub Notifications Snoozer</title>
+    <link rel="stylesheet" type="text/less" href="app.less">
```
Comment (probablycorey) [inline src/index.html]: Maybe this should be `src/app.less` or `./src/app.less`?

File: src/main.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ function createWindow () {
   // and load the index.html of the app.
   mainWindow.loadURL(`file://${__dirname}/index.html`)
 
-  // mainWindow.webContents.openDevTools()
+  mainWindow.webContents.openDevTools()
```
Comment (probablycorey) [inline src/main.js]: **protip:** You can press `cmd+opt+i` to open up the console

File: .compilerc (hunk 1 above)
Comment (cheshire137) [inline .compilerc:1]: Yep! See https://github.com/electron/electron-compile#how-do-i-set-up-babel--less--whatever-the-way-i-want

File: src/index.html (hunk 2 above)
Comment (cheshire137) [inline src/index.html]: Hmm, just tried both, no luck. Still a white background everywhere!

File: src/main.js (hunk 3 above)
Comment (cheshire137) [inline src/main.js]: Ah yeah, I was doing that then got lazy. I'll comment this back out, didn't intend to check it in!

**Comment by cheshire137** [comment]
I spoke with @zeke about this yesterday, he thought he might be able to help!

**Comment by zeke** [comment]
👋 I haven't figured out the less issue yet, but I did notice that `electron` is in the dependencies, which could be screwing things up. That package name is a different project. Fortunately, that user has graciously handed over the `electron` npm name to us, but we haven't yet published something like `electron-prebuilt` in its place.

**Comment by cheshire137** [comment]
Hm, I got this branch updated with master just now and `electron` is no longer a dependency in package.json.

**Comment by zeke** [comment]
The Less configuration seems right, but I'm not seeing the styles.

I tried debugging with `DEBUG=*,-babel npm start`, but didn't see any issues in the logs.

@paulcbetts do you see any obvious signs of misconfiguration here?

**Comment by anaisbetts** [comment]
I'll have a look in a hot second

File: src/components/App/App.less [hunk 4]
```diff
@@ -0,0 +1,5 @@
+@import "../Filter/Filter.less";
```
Comment (anaisbetts) [inline src/components/App/App.less:1]: It's probably easier to just include this inline in the JSX file:

``` html
<style type="text/less">
@import '../filter/filter.less';
...
</style>
```

**Comment by anaisbetts** [comment]
@cheshire137 Ok, tracked this down, it turns out I am not good at computer

**Comment by anaisbetts** [comment]
Ok! Blow away `node_modules` and reinstall so that you get `electron-compilers@3.1.6` and this should just work. You should also go into $TMPDIR and remove any folder that starts with `compileCache`

**Comment by cheshire137** [comment]
Thank you, it works!

screen shot 2016-07-20 at 1 22 50 pm

I checked out [your commits](https://github.com/electron/electron-compilers/compare/8b77d4cb5ed5c3843f6efcc3fb158f442e053f92...3d9498f2dbd7da1e7c64605d0235ce3101f572ae)--thanks for helping!

**Comment by zeke** [comment]
👏

File: package.json [hunk 5]
```diff
@@ -12,13 +12,19 @@
     "electron-prebuilt-compile": "^1.2.5",
     "eslint": "^3.0.1",
     "eslint-config-airbnb": "^9.0.1",
+    "eslint-plugin-import": "^1.11.0",
+    "eslint-plugin-jsx-a11y": "^2.0.1",
+    "eslint-plugin-react": "^5.2.2",
```
Comment (probablycorey) [inline package.json:17]: Do we need these three packages? I think they are included with the `eslint-config-airbnb` package's dependencies.

Comment (cheshire137) [inline package.json:17]: We do! Check the full commit messages, tests failed on Travis till these were added.

Comment (anaisbetts) [inline package.json:17]: Is Travis running old-ass npm 2 or something?

Comment (cheshire137) [inline package.json:17]: Haha I think my Travis config file specifies node 6. Check a build, it spits out node, npm, and nvm version numbers.

**Pull request merged**